### ISSUE TITLE
Error in AutoCatBoostRegression

### ISSUE THREAD

**Issue by mbanco** [AdrianAntico/RemixAutoML#67]
Hi Adrian,
the following code throws an error:

options(warn=-1);
options(echo=FALSE);
options(verbose=FALSE);

y <- c(27978296.87, 26450099.1, 26330876.23, 26549732.37, 25882933.74, 26710469, 24964791, 28029292.7, 28376283.74, 31250598.35, 29777293.28, 32103699.2, 31422163.29, 31904695.36, 31256477.22, 32933981, 30943126.24, 36154304.51, 34195795.65, 37555676.01, 38792391.57, 41205297.9, 37770409.12, 40163566.37, 39913420.91, 39883615.4, 39830468.01, 44365814.9, 40424157.14);
X <- as.matrix(structure(c(2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 448823000, 452582049, 447265388, 490861113, 513144605.1, 536342069.6, 527382672, 550946233.5, 575264038.3, 603909910, 607605882.6, 652517503.2, 666332636.9, 668311223.6, 678146483.8, 713702671.4, 710798424.4, 782363379.2, 763253115.9, 799363641.2, 833472705.8, 896614149.8, 910416441.4, 926245640.6, 1001489314, 954231655, 997926895, 1075596578, 1020595851, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 487826716, 448823000, 452582049, 447265388, 490861113, 513144605.1, 536342069.6, 527382672, 550946233.5, 575264038.3, 603909910, 607605882.6, 652517503.2, 666332636.9, 668311223.6, 678146483.8, 713702671.4, 710798424.4, 782363379.2, 763253115.9, 799363641.2, 833472705.8, 896614149.8, 910416441.4, 926245640.6, 1001489314, 954231655, 997926895, 1075596578), .Dim = c(29, 4)));
X_fut <- as.matrix(structure(c(1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1140777731, 1082443960, 1208882161, 1147065864, 1280085320, 1214628043, 1354458278, 1285197933, 1432204183, 1358968294, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 1020595851, 1140777731, 1082443960, 1208882161, 1147065864, 1280085320, 1214628043, 1354458278, 1285197933, 1432204183), .Dim = c(10, 4)));

paq <- (.packages());

if (sum((paq == 'RemixAutoML') == TRUE) == 0) {
  
  library(RemixAutoML)};

rm(paq);

datos <- as.data.frame(cbind(y, X));

nombre_col <- paste('X', 1:NCOL(X), sep='');

colnames(datos) <- c('y', nombre_col);

if (NCOL(X)==1) {
  
  FeatureCol <- 2;
  
} else {
  
  FeatureCol <- 2:(NCOL(X)+1);
  
};

set.seed(42);

fit_CatBoost <- AutoCatBoostRegression(
  # GPU or CPU and the number of available GPUs
  task_type = 'CPU',
  ModelID = 'FirstModel',
  model_path = normalizePath('./'),
  metadata_path = NULL,
  SaveModelObjects = FALSE,
  ReturnModelObjects = TRUE,
  
  data = datos,
  TrainOnFull = FALSE,
  ValidationData = NULL,
  TestData = NULL,
  TargetColumnName = 'y',
  FeatureColNames = FeatureCol,
  PrimaryDateColumn = NULL,
  IDcols = NULL,
  TransformNumericColumns = NULL,
  Methods = c('BoxCox', 'Asinh', 'Asin', 'Log', 'LogPlus1', 'Logit', 'YeoJohnson'),
  
  eval_metric = 'RMSE',
  loss_function = 'RMSE',
  MetricPeriods = 10L,
  NumOfParDepPlots = ncol(datos)-1L-2L,
  EvalPlots = FALSE,
  
  PassInGrid = NULL,
  GridTune = TRUE,
  MaxModelsInGrid = 50L,
  MaxRunsWithoutNewWinner = 100L, 
  MaxRunMinutes = 60*60,
  Shuffles = 4L,
  BaselineComparison = 'default',
  
  Trees = 100L,
  Depth = NULL, 
  LearningRate = NULL,
  L2_Leaf_Reg = NULL, 
  RSM = NULL,
  BootStrapType = c('Bayesian', 'Bernoulli', 'Poisson', 'MVS', 'No'),
  GrowPolicy = c('SymmetricTree', 'Depthwise', 'Lossguide'));

This is the error:

Error in catboost::catboost.train(learn_pool = TrainPool, test_pool = TestPool,  : 
  c:/program files (x86)/go agent/pipelines/buildmaster/catboost.git/catboost/private/libs/options/json_helper.h:156: Can't parse parameter "learning_rate" with value: {}
Timing stopped at: 0 0 0

Best Regards.

**Comment by AdrianAntico** [comment]
@mbanco When you set GridTune = TRUE you cannot supply NULL as values for the hyperparameters 'Trees', 'Depth', 'LearningRate', 'L2_Leaf_Reg', 'RSM', 'BootStrapType', or 'GrowPolicy'. Those are all used in bandit grid tuning. 

I just added some code to the beginning of that function that will check for NULL values and if they are NULL when GridTune = TRUE the function will immediately return a note indicating which variables was NULL. 

Also, I would use the values I supply in the help file example as a starting point for assigning values to those parameters.
